src/Domain/LanguageCodes.php

```diff
@@ -2,32 +2,65 @@
 
 namespace SandwaveIo\RealtimeRegister\Domain;
 
-final class LanguageCodes implements DomainObjectInterface
+use ArrayAccess;
+use Countable;
+
+final class LanguageCodes implements ArrayAccess, Countable
 {
-    public string $name;
-    public ?string $allowedCharacters;
+    /** @var array<string, LanguageCode> */
+    public array $entities;
 
-    private function __construct(string $name, ?string $allowedCharacters)
+    private function __construct(array $entities)
     {
-        $this->name = $name;
-        $this->allowedCharacters = $allowedCharacters;
+        $this->entities = $entities;
     }
 
-    public static function fromArray(array $json): LanguageCodes
+    public static function fromArray(array $json): self
     {
-        return new LanguageCodes(
-            $json['name'],
-            $json['allowedCharacters'] ?? null
-        );
+        $entities = [];
+
+        foreach ($json as $countryCode => $languageCode) {
+            $entities[$countryCode] = LanguageCode::fromArray($languageCode);
+        }
+
+        return new self($entities);
     }
 
     public function toArray(): array
     {
-        return array_filter([
-            'name'              => $this->name,
-            'allowedCharacters' => $this->allowedCharacters,
-        ], function ($x) {
-            return ! is_null($x);
-        });
+        $languageCodesArray = [];
+
+        foreach ($this->entities as $countryCode => $languageCode) {
+            $languageCodesArray[$countryCode] = $languageCode->toArray();
+        }
+
+        return $languageCodesArray;
+    }
+
+    public function offsetExists($offset): bool
```

Comment:
What is offeset?

Reply:
This is because of the ArrayAccess interface so we can use the LanguageCodes object as an array.